src/utils/cloning.ts

```diff
@@ -480,6 +502,30 @@ function cloneASTNodeImpl<T extends ASTNode>(
       [...node.vParameters].map((p) => cloneASTNodeImpl(p, ast, remappedIds)),
       node.raw,
     );
+  } else if (node instanceof ContractDefinition) {
```

Comment:
I think declarations such as `ContractDefinition` and `EnumDefinition` should go in the previous section with the other ones, and leave `ParameterList` alone in `//ASTNodeWithChildren`